sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -1492,6 +1491,8 @@ class Dataset[T] private[sql](
    * @param weights weights for splits, will be normalized if they don't sum to 1.
    * @param seed Seed for sampling.
    *
+   * For Java API, use [[randomSplitAsList]].
```

Comment:
Dataset.randomSplit() was returning Array[Dataset[T]], which doesn't work for Java users since Java methods can't return generic arrays. Hence randomSplitAsList().